packages/nodes-base/nodes/Mattermost/v1/actions/router.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	IExecuteFunctions,
+} from 'n8n-core';
+
+import {
+	INodeExecutionData, NodeApiError,
+} from 'n8n-workflow';
+
+import * as channel from './channel';
+import * as message from './message';
+import * as reaction from './reaction';
+import * as user from './user';
+
+export async function router(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+	const items = this.getInputData();
+	const operationResult: INodeExecutionData[] = [];
+
+	for (let i = 0; i < items.length; i++) {
+		const resource = this.getNodeParameter('resource', i) as string;
+		let operation = this.getNodeParameter('operation', i) as string;
+		if (operation === 'delete') {
+			operation = 'del';
+		} else if (operation === 'desactive') {
+			operation = 'deactive';
+		}
+		const mattermost = {
+			resource,
+			operation,
+		} as Mattermost;
+		
+		try {
+			if (mattermost.resource === 'channel') {
+				operationResult.push(...await channel[mattermost.operation].call(this, i));
+			} else if (mattermost.resource === 'message') {
+				operationResult.push(...await message[mattermost.operation].call(this, i));
+			} else if (mattermost.resource === 'reaction') {
+				operationResult.push(...await reaction[mattermost.operation].call(this, i));
+			} else if (mattermost.resource === 'user') {
+				operationResult.push(...await user[mattermost.operation].call(this, i));
+			} else {
+				throw new NodeApiError(this.getNode(), {message: 'Resource not supported.'});
```

Comment:
1. `NodeOperationError` instead of `NodeApiError`. 
2. In fact, at this `else` clause we have exhausted the union (if you try to access `mattermost.resource`, its type will be `never`) so instead of catching the error, we can rely on the IDE/TS to disallow us from using a non-existing resource.